### PR TITLE
Support Neovim plugins (and basic .vimrc support)

### PR DESCRIPTION
**Pull request by Chillee**
A basic proof of concept. Many things need to be added and made more general before this is ready to push, but I think it's pretty cool.

Refer to https://github.com/Chillee/Vim/blob/neovimPlugins/README.md#plugins for the plugins I've verified work so far (ie: just targets.vim so far).

### CODE REVIEW

File: src/neovim/nvimUtil.ts [hunk 1]
```diff
@@ -9,11 +9,16 @@ import { spawn } from "child_process";
...
+import * as os from "os";
+
+type UndoTree = {entries: Array<{seq: number, time: number}>, save_cur: number, save_last: number, seq_cur: number, seq_last: number, synced: number, time_cur: number};
 
 export class Neovim {
 
   static async initNvim(vimState: VimState) {
-    const proc = spawn(Configuration.neovimPath, ['-u', 'NONE', '-N', '--embed'], { cwd: __dirname });
+    this.syncVSSettingsToVim();
+    const proc = spawn(Configuration.neovimPath, ['-u', '~/.config/nvim/init.vim', '-N', '--embed'], { cwd: __dirname });
```
Comment (justinmk) [inline src/neovim/nvimUtil.ts]: could omit the `-u ...` entirely so that Nvim looks in the usual places (on Windows also)

Comment (Chillee) [inline src/neovim/nvimUtil.ts]: I haven't added the logic yet, but I think the neovim embedding in VSCode should use a custom init.vim file. We're going to need to do a lot of autogeneration of the init.vim file for various setting sync reasons, and I'd rather not be overwriting users' original settings.

**Comment by Chillee** [comment]
This will probably be a no go, in favor of the full neovim integration.

**Comment by Chillee** [comment]
As I'm nearly certain this is the way to go for neovim integration: https://github.com/VSCodeVim/Vim/pull/1897, I will finally be closing this PR.

**Pull request closed without merging**